package/src/internal/types.ts

```diff
@@ -28,4 +28,6 @@ export type Prettify<T> = {
 	[K in keyof T]: T[K];
 } & {};
 
+export type ExtendedPrettify<T> = T extends infer U ? Prettify<U> : never;
```

Comment:
what does this do?

Reply:
It is a TS trick to force a generic to expand to its component types.